packages/ui/components/ui/tx/view/value.tsx

```diff
@@ -1,21 +1,90 @@
-import { ValueView } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/asset/v1/asset_pb';
+import { Metadata, ValueView } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/asset/v1/asset_pb';
 import {
   fromBaseUnitAmount,
   getDisplayDenomExponent,
   getDisplayDenomFromView,
 } from '@penumbra-zone/types';
 import { CopyToClipboard } from '../../copy-to-clipboard';
+//import { AssetIcon } from '../../webapp/shared/asset-icon.tsx';
 import { CopyIcon } from '@radix-ui/react-icons';
 import { Amount } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/num/v1/num_pb';
+import { Identicon } from '../../identicon';
 
 interface ValueViewProps {
   view: ValueView | undefined;
   showDenom?: boolean;
+  showValue?: boolean;
+  showIcon?: boolean;
+  showEquivalent?: boolean;
 }
 
-export const ValueViewComponent = ({ view, showDenom = true }: ValueViewProps) => {
+export const AssetIcon = ({ metadata }: { metadata?: Metadata }) => {
+  const icon = metadata?.images[0]?.png || metadata?.images[0]?.svg;
+  return (
+    <>
+      {icon ? (
+        <img className='size-6 rounded-full' src={icon} alt='Asset icon' />
+      ) : (
+        <Identicon
+          uniqueIdentifier={metadata?.symbol ?? '?'}
+          size={24}
+          className='rounded-full'
+          type='solid'
+        />
+      )}
+    </>
+  );
+};
+
+
+export const ValueViewComponent = ({ view, showDenom = true, showValue = true, showIcon = true, showEquivalent = true }: ValueViewProps) => {
   if (!view) return <></>;
 
+  if (view.valueView.case === 'knownAssetId' && view.valueView.value.metadata) {
+    const value = view.valueView.value;
+    const metadata = view.valueView.value.metadata;
+    const amount = value.amount ?? new Amount();
+    const exponent = getDisplayDenomExponent(metadata);
+    const formattedAmount = fromBaseUnitAmount(amount, exponent).toFormat();
```

Comment:
Ideally this would round the number of decimal places to something reasonable (see the USD screenshot) but I wasn't sure how to do this, passing "6" to `.toFormat()` gave `.000000` even for round numbers.